File: c++/dr.cpp
#include <iostream>
#include <cmath>
using namespace std;

#define PI 3.14159

// 1. Assign values program
void program1() {
    int a = 4, b = 3, c = 2;
    cout << "Program 1: " << a << ", " << b << ", " << c << endl;
}

// 2. Print message program
void program2() {
    cout << "Program 2: C language is a powerful programming language." << endl;
}

// 3. Interchange two variables program
void program3() {
    int a = 3, b = 5, temp;
    temp = a; 
    a = b; 
    b = temp;
    cout << "Program 3: a=" << a << ", b=" << b << endl;
}

// 4. Year → months program
void program4() {
    int year = 19;
    cout << "Program 4: Months = " << year * 12 << endl;
}

// 5. Sum and product program
void program5() {
    int x, y;
    cout << "Enter any two numbers: ";
    cin >> x >> y;
    cout << "Program 5: Sum=" << x + y << endl;
    cout << "Product=" << x * y << endl;
}

// 6. Age in months program
void program6() {
    int age;
    cout << "Enter your age: ";
    cin >> age;
    cout << "Program 6: Age in months = " << age * 12 << endl;
}

// 7. Roll No + 3 subject marks program
void program7() {
    int roll, cg, ic, pf;
    cout << "Enter your roll number: ";
    cin >> roll;

    cout << "Enter your marks in pf: ";
    cin >> pf;

    cout << "Enter your marks in cg: ";
    cin >> cg;

    cout << "Enter your marks in ic: ";
    cin >> ic;

    int total = cg + ic + pf;
    float avg = total / 3.0;

    cout << "Program 7: Total=" << total << endl;
    cout << "Average=" << avg << endl;
}

// 8. Fahrenheit to Celsius program
void program8() {
    float f;
    cout << "Enter temperature (in Fahrenheit): ";
    cin >> f;
    float c = (5.0 / 9) * (f - 32);
    cout << "Program 8: Celsius = " << c << endl;
}

// 9. Max of 4 numbers program
void program9() {
    int a = 1, b = 2, c = 5, d = 8;
    int max = a;

    if (b > max) max = b;
    if (c > max) max = c;
    if (d > max) max = d;

    cout << "Program 9: The Maximum number is: " << max << endl;
}

// 10. Convert miles to km program
void program10() {
    float miles = 2.5;
    cout << "Program 10: Km = " << miles * 1.609 << endl;
}

// 11. Average of 2 ints program
void program11() {
    int a = 5, b = 9;
    cout << "Program 11: Average = " << (a + b) / 2.0 << endl;
}

// 12. Volume of cylinder program
void program12() {
    const float pi = 3.14159;
    float r = 3, h = 5;
    cout << "Program 12: Volume = " << pi * r * r * h << endl;
}

// 13. Area program
void program13() {
    float r = 3;
    cout << "Program 13: Area = " << PI * r * r << endl;
}

// 14. mm to inches program
void program14() {
    float mm = 20;
    cout << "Program 14: Inches = " << mm / 25.4 << endl;
}

// 15. Interchange values program
void program15() {
    int a = 5, b = 8, temp;
    temp = a;
    a = b;
    b = temp;
    cout << "Program 15: a=" << a << ", b=" << b << endl;
}

// 16. a=b=c=3 product program
void program16() {
    int a = 3, b = 3, c = 3;
    cout << "Program 16: Product = " << a * b * c << endl;
}

// 17. Separate integer & fraction part program
void program17() {
    float x = 15.58971;
    int i = int(x);
    float f = x - i;
    cout << "Program 17: Int=" << i << ", Fraction=" << f << endl;
}

// 18. s = vi*t + 1/2 at^2 program
void program18() {
    float vi = 5.0, t = 2.0, a = 3.0;
    float s = vi * t + 0.5 * a * t * t;
    cout << "Program 18: s=" << s << endl;
}

// 19. Age in months and days program
void program19() {
    int age = 12;
    cout << "Program 19: Your age is "<<age<<" and month=" << age * 12 << ", Days=" << age * 365 << endl;
}

// 20. Escape sequence printing program
void program20() {
    cout << "Program 20:\nC:\\Windows>\n'P'\t'A'\t'K'\n\"Pakistan\"\n";
}

// 21. Print pattern with \n in one statement program
void program21() {
    cout << "Program 21:\nXXXXX\nXXXX\nXXX\nXX\nX\n";
}

// 22. Input student details program
void program22() {
    string name;
    int age;
    float height;
    char gender;

    cout << "Enter your name: ";
    cin >> name;

    cout << "Enter your age: ";
    cin >> age;

    cout << "Enter your height: ";
    cin >> height;

    cout << "Enter your gender (M/F): ";
    cin >> gender;

    cout << "Program 22: " << name << ", " << age << ", " << height << ", " << gender << endl;
}

// 23. Circle area & circumference program
void program23() {
    float r;
    cout << "Enter the radius: ";
    cin >> r;
    cout << "Program 23: Area=" << PI * r * r << ", Circum=" << 2 * PI * r << endl;
}

// 24. Total & average of 5 subjects program
void program24() {
    int icp, cag, fe, pf, ap;

    cout << "Enter marks of CAG: ";
    cin >> cag;

    cout << "Enter marks of ICP: ";
    cin >> icp;

    cout << "Enter marks of PF: ";
    cin >> pf;

    cout << "Enter marks of FE: ";
    cin >> fe;

    cout << "Enter marks of AP: ";
    cin >> ap;

    int total = ap + fe + cag + icp + pf;
    cout << "Program 24: Total=" << total << ", Average=" << total / 5.0 << endl;
}

// 25. Area of triangle program (Heron's formula)
void program25() {
    double a, b, c;
    cout << "Enter value of a: ";
    cin >> a;
    cout << "Enter value of b: ";
    cin >> b;
    cout << "Enter value of c: ";
    cin >> c;

    double s = (a + b + c) / 2;
    double area = sqrt(s * (s - a) * (s - b) * (s - c));

    cout << "Program 25: Area=" << area << endl;
}

// 26. Rupees to dollars program
void program26() {
    float rs = 12000;
    cout << "Program 26: Dollars = " << rs / 60 << endl;
}

// 27. Convert time to seconds program
void program27() {
    int h, m, s;
    cout << "Enter hours: ";
    cin >> h;
    cout << "Enter minutes: ";
    cin >> m;
    cout << "Enter seconds: ";
    cin >> s;

    cout << "Program 27: Seconds = " << h * 3600 + m * 60 + s << endl;
}

// 28. disc program
void program28() {
    int a, b, c;
    cout << "Enter value of a: ";
    cin >> a;
    cout << "Enter value of b: ";
    cin >> b;
    cout << "Enter value of c: ";
    cin >> c;

    cout << "Program 28: disc = " << b * b - 4 * a * c << endl;
}

// 29. If n > 100 program
void program29() {
    int n;
    cout << "Enter value of n: ";
    cin >> n;

    if (n > 100)
        cout << "Program 29: n is greater than 100" << endl;
}

// 30. Odd or even program
void program30() {
    int n;
    cout << "Enter value of n: ";
    cin >> n;

    if (n % 2 == 0)
        cout << "Program 30: Even\n";
    else
        cout << "Program 30: Odd\n";
}

void showMenu() {

        int choice;

        cout << "\n=== MENU ===\n";
        cout<<"01. Program1 \n";
    cout<<"02. Program2 \n";
    cout<<"03. Program3 \n";
    cout<<"04. Program4 \n";
    cout<<"05. Program5 \n";
    cout<<"06. Program6 \n";
    cout<<"07. Program7 \n";
    cout<<"08. Program8 \n";
    cout<<"09. Program9 \n";
    cout<<"10. Program10 \n";
    cout<<"11. Program11 \n";
    cout<<"12. Program12 \n";
    cout<<"13. Program13 \n";
    cout<<"14. Program14 \n";
    cout<<"15. Program15 \n";
    cout<<"16. Program16 \n";
    cout<<"17. Program17 \n";
    cout<<"18. Program18 \n";
    cout<<"19. Program19 \n";
    cout<<"20. Program20 \n";
    cout<<"21. Program21 \n";
    cout<<"22. Program22 \n";
    cout<<"23. Program23 \n";
    cout<<"24. Program24 \n";
    cout<<"25. Program25 \n";
    cout<<"26. Program26 \n";
    cout<<"27. Program27 \n";
    cout<<"28. Program28 \n";
    cout<<"29. Program29 \n";
    cout<<"30. Program30 \n";
    cout<<"----------Exit-----------";
        cout << "Enter choice: ";
        cin >> choice;

        switch (choice) {
            case 1: program1(); break;
            case 2: program2(); break;
            case 3: program3(); break;
            case 4: program4(); break;
            case 5: program5(); break;
            case 6: program6(); break;
            case 7: program7(); break;
            case 8: program8(); break;
            case 9: program9(); break;
            case 10: program10(); break;
            case 11: program11(); break;
            case 12: program12(); break;
            case 13: program13(); break;
            case 14: program14(); break;
            case 15: program15(); break;
            case 16: program16(); break;
            case 17: program17(); break;
            case 18: program18(); break;
            case 19: program19(); break;
            case 20: program20(); break;
            case 21: program21(); break;
            case 22: program22(); break;
            case 23: program23(); break;
            case 24: program24(); break;
            case 25: program25(); break;
            case 26: program26(); break;
            case 27: program27(); break;
            case 28: program28(); break;
            case 29: program29(); break;
            case 30: program30(); break;

            

            default:
                cout << "Invalid choice! Try again.\n";
        }
    }


int main() {
    showMenu();
    return 0;
}

File: python/dr.py
PI = 3.14159

# 1
def program1():
    a, b, c = 4, 3, 2
    print("Program 1:", a, b, c)

# 2
def program2():
    print("Program 2: C language is a powerful programming language.")

# 3
def program3():
    a, b = 3, 5
    a, b = b, a
    print("Program 3:", a, b)

# 4
def program4():
    year = 19
    print("Program 4: Months =", year * 12)

# 5
def program5():
    x = int(input("Enter number 1: "))
    y = int(input("Enter number 2: "))
    print("Sum =", x + y)
    print("Product =", x * y)

# 6
def program6():
    age = int(input("Enter your age: "))
    print("Age in months =", age * 12)

# 7
def program7():
    roll = int(input("Roll no: "))
    pf = int(input("Marks in PF: "))
    cg = int(input("Marks in CG: "))
    ic = int(input("Marks in IC: "))
    total = pf + cg + ic
    avg = total / 3
    print("Total =", total)
    print("Average =", avg)

# 8
def program8():
    f = float(input("Enter Fahrenheit: "))
    c = (5/9) * (f - 32)
    print("Celsius =", c)

# 9
def program9():
    a, b, c, d = 1, 2, 5, 8
    print("Maximum =", max(a, b, c, d))

# 10
def program10():
    miles = 2.5
    print("Km =", miles * 1.609)

# 11
def program11():
    a, b = 5, 9
    print("Average =", (a + b) / 2)

# 12
def program12():
    r, h = 3, 5
    volume = PI * r * r * h
    print("Volume =", volume)

# 13
def program13():
    r = 3
    print("Area =", PI * r * r)

# 14
def program14():
    mm = 20
    print("Inches =", mm / 25.4)

# 15
def program15():
    a, b = 5, 8
    a, b = b, a
    print("a =", a, "b =", b)

# 16
def program16():
    a = b = c = 3
    print("Product =", a*b*c)

# 17
def program17():
    x = 15.58971
    i = int(x)
    f = x - i
    print("Int =", i, "Fraction =", f)

# 18
def program18():
    vi, t, a = 5, 2, 3
    s = vi*t + 0.5*a*t*t
    print("s =", s)

# 19
def program19():
    age = 19
    print("Your age is: ",age," Months =", age * 12, "Days =", age * 365)

# 20
def program20():
    print("C:\\Windows>")
    print("'P'\t'A'\t'K'")
    print("\"Pakistan\"")

# 21
def program21():
    print("XXXXX\nXXXX\nXXX\nXX\nX")

# 22
def program22():
    name = input("Name: ")
    age = int(input("Age: "))
    height = float(input("Height: "))
    gender = input("Gender M/F: ")
    print(name, age, height, gender)

# 23
def program23():
    r = float(input("Enter radius: "))
    print("Area =", PI*r*r, "Circum =", 2*PI*r)

# 24
def program24():
    cag = int(input("CAG: "))
    icp = int(input("ICP: "))
    pf = int(input("PF: "))
    fe = int(input("FE: "))
    ap = int(input("AP: "))
    total = cag + icp + pf + fe + ap
    print("Total =", total, "Average =", total/5)

# 25
def program25():
    import math
    a = float(input("a: "))
    b = float(input("b: "))
    c = float(input("c: "))
    s = (a + b + c) / 2
    area = math.sqrt(s*(s-a)*(s-b)*(s-c))
    print("Area =", area)

# 26
def program26():
    rs = 12000
    print("Dollars =", rs/60)

# 27
def program27():
    h = int(input("Hours: "))
    m = int(input("Minutes: "))
    s = int(input("Seconds: "))
    print("Total seconds =", h*3600 + m*60 + s)

# 28
def program28():
    a = int(input("a: "))
    b = int(input("b: "))
    c = int(input("c: "))
    print("Disc =", b*b - 4*a*c)

# 29
def program29():
    n = int(input("n: "))
    if n > 100:
        print("n is greater than 100")

# 30
def program30():
    n = int(input("Enter number: "))
    if n % 2 == 0:
        print("Even")
    else:
        print("Odd")


# MENU
def showMenu():
    print("------ MENU ------")
    print("1. Program 1")
    print("2. Program 2")
    print("3. Program 3")
    print("4. Program 4")
    print("5. Program 5")
    print("6. Program 6")
    print("7. Program 7")
    print("8. Program 8")
    print("9. Program 9")
    print("10. Program 10")
    print("11. Program 11")
    print("12. Program 12")
    print("13. Program 13")
    print("14. Program 14")
    print("15. Program 15")
    print("16. Program 16")
    print("17. Program 17")
    print("18. Program 18")
    print("19. Program 19")
    print("20. Program 20")
    print("21. Program 21")
    print("22. Program 22")
    print("23. Program 23")
    print("24. Program 24")
    print("25. Program 25")
    print("26. Program 26")
    print("27. Program 27")
    print("28. Program 28")
    print("29. Program 29")
    print("30. Program 30")

    choice = int(input("Enter your choice: "))

    if choice == 1:
        program1()
    elif choice == 2:
        program2()
    elif choice == 3:
        program3()
    elif choice == 4:
        program4()
    elif choice == 5:
        program5()
    elif choice == 6:
        program6()
    elif choice == 7:
        program7()
    elif choice == 8:
        program8()
    elif choice == 9:
        program9()
    elif choice == 10:
        program10()
    elif choice == 11:
        program11()
    elif choice == 12:
        program12()
    elif choice == 13:
        program13()
    elif choice == 14:
        program14()
    elif choice == 15:
        program15()
    elif choice == 16:
        program16()
    elif choice == 17:
        program17()
    elif choice == 18:
        program18()
    elif choice == 19:
        program19()
    elif choice == 20:
        program20()
    elif choice == 21:
        program21()
    elif choice == 22:
        program22()
    elif choice == 23:
        program23()
    elif choice == 24:
        program24()
    elif choice == 25:
        program25()
    elif choice == 26:
        program26()
    elif choice == 27:
        program27()
    elif choice == 28:
        program28()
    elif choice == 29:
        program29()
    elif choice == 30:
        program30()
    else:
        print("Invalid choice!")

showMenu()